### PR TITLE
va-file-input: add support for read-only mode

### PR DESCRIPTION
**Pull request by pennja**
## Chromatic
https://file-input-field-updates-01--65a6e2ed2314f7b8f98609d8.chromatic.com

## Description
Closes [[File input] Read only variation - Development #2960](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2960)

## QA Checklist
- [x] Component maintains 1:1 parity with design mocks
- [x] Text is consistent with what's been provided in the mocks
- [x] Component behaves as expected across breakpoints
- [x] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [x] Tab order and focus state work as expected
- [x] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [x] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [x] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots

(See latest screenshots in comments below.)

## Acceptance criteria
- [x] QA checklist has been completed
- [x] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by micahchiang — COMMENTED** [review]
Hey @pennja! Thanks for working on this task, we really appreciate it. I've tested your solution in storybook and it looks fine to me. Would you mind adding a story for the read-only version of file-input in storybook? It can be included in this PR since our instance of storybook is a package that's collocated with web-components. 

Could you also include some screenshots in this PR that portray the changes you've made? Thanks!

**Comment by jamigibbs** [comment]
@babsdenney I added you for a design review because I think you were the one that [added it to Figma](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2877)? Feel free to hot potato this to someone else if I got it wrong.

**Review by pennja — COMMENTED** [review]
@micahchiang I was able to make the changes you asked for but I'm running into a chromatic build error. If anyone has any suggestions on how to troubleshoot this, I'd really appreciate the guidance!

**Comment by jamigibbs** [comment]
@pennja There specific Storybook error is:

```
WARN 🚨 Unable to index ./stories/va-file-input-uswds.stories.jsx:
WARN   SyntaxError: Unexpected token, expected "," (35:2)
```

This is line 35 in the storybook file:

```
  'uswds': true,
  'header-size': null,
  'children': null <------
  'value': null,
  'children': null,
};
```

It looks to me like you're missing a comma.

**Comment by ataker** [comment]
@pennja Looks like you were just missing a comma. I pushed an update.

**Review by micahchiang — APPROVED** [review]
Looks good to me. Thanks for the contribution @pennja! Last request is that you wait for an approval from @babsdenney or @danbrady before merging in.

**Comment by danbrady** [comment]
Thanks for jumping in, @pennja! I noticed a few things listed below. Attaching a screenshot for reference:

image

* There shouldn't be any space between the file name and weight. Looks like an empty error `span`.
* Because this is the read-only version, the box label should be "Files you uploaded" not "Select a file". This language came from collaborating with CAIA.
* I noticed I can click the "Select a file to upload" label and it opens the file browser prompt like a regular file input. Because this is the read-only version, we shouldn't have that behavior. In fact, we don't show the "Select a file to upload" and hint text, just the blue box.
* Would it be possible to show read-only examples that have an additional (read-only) form inputs and/or multiple files? (You can see multiple files in our [examples file](https://www.figma.com/design/JDFpGLIojfuQwANXScQjqe/VADS-Component-Examples?node-id=4264-7410&t=bGVsLcbGoIMhl1ch-1). We should create a Figma example that shows additional read-only form inputs.) Attached shots from there:



<br>

<br>



@babsdenney I think you worked in the design of this variation. Would you mind taking a look as well to see if I missed anything?

@pennja This is great work. Thanks again for contributing!

**Comment by babsdenney** [comment]
Thanks @jamigibbs and @danbrady I'm going to take a look right now.

**Comment by babsdenney** [comment]
@danbrady This was a great review! Thank you for doing that, I think I would have missed the clickable area that still allowed you to select files. Thank for finding that and agree that it would be nice to see the other variations with the additional form fields and multiple file options. Thank you @pennja for submitting this and contributing.

**Comment by micahchiang** [comment]
Hey @pennja! I wanted to bump this PR and check whether or not you've seen the feedback @danbrady and @babsdenney have left.

File: packages/web-components/src/components/va-file-input/va-file-input.tsx
Comment (it-harrison) [inline packages/web-components/src/components/va-file-input/va-file-input.tsx:13]: this will probably result in this component not rendering translations if this component is installed independently in an application, e.g. outside of vets-website

**Comment by Andrew565** [comment]
Implemented the requested design changes and updated the storybook examples.
Screenshot 2024-10-10 at 3 54 42 PM
Next up will be porting these changes to the va-file-input-multiple component.

File: packages/storybook/stories/va-file-input-multiple-uswds.stories.jsx
Comment (Andrew565) [inline packages/storybook/stories/va-file-input-multiple-uswds.stories.jsx:44]: Renamed "additional" prop to "children" to be in line with va-file-input (single).

**Comment by Andrew565** [comment]
Added va-file-input-multiple read-only support.

Screenshot 2024-10-18 at 11 58 46 AM

Screenshot 2024-10-18 at 11 59 09 AM

Screenshot 2024-10-18 at 11 59 16 AM

**Comment by danbrady** [comment]
Thanks for the updates, @Andrew565. Some minor visual issues:

image

* File cards should all align (note the red lines)
* When there is not an additional form field, there should not be a separator line. When the line is removed, keep the spacing around the item consistent.
* Can the change/delete buttons stay aligned on the same line, but wrap/stack only when they don't fit side-by-side?  
* When a read-only version has an additional form field, that form should be read-only as well. This variation is used on form review pages, so all additional fields should be previously set. (Not sure if this is just a storybook example or not.)


Here's what it should look like:
image

**Comment by Andrew565** [comment]
@danbrady @babsdenney Design fixes in place for va-file-input and va-file-input-multiple.

Screenshot 2024-10-22 at 12 34 39 PM

Screenshot 2024-10-22 at 12 34 56 PM

Screenshot 2024-10-22 at 12 37 08 PM

**Comment by danbrady** [comment]
@Andrew565 Looking much better. One minor issue I commented in Chromatic about the space missing between the separator lines and the buttons. (Easier to see when hovering.) Adding screens here:

Previously:
Screenshot 2024-10-22 at 3 51 59 PM

Latest:
Screenshot 2024-10-22 at 3 50 55 PM

**Comment by Andrew565** [comment]
@danbrady Fixed the margin-top for the file button section. Set it to 16px.

File: packages/web-components/src/components/va-file-input/va-file-input.scss
Comment (danbrady) [inline packages/web-components/src/components/va-file-input/va-file-input.scss:146]: @Andrew565 Let's make it 8px instead of 16px, please. Otherwise, all looks good.

**Review by danbrady — APPROVED** [review]

**Comment by Andrew565** [comment]
@ataker et al, this is ready for re-review, thanks!

**Pull request merged**
